converse.js

```diff
@@ -634,11 +635,35 @@
             },this), 200));
         };
 
+        this.ping = function (jid, timeout, success, error){
+            if (typeof jid === 'undefined'     || jid == null)     { jid = this.bare_jid; }
+                if (typeof timeout === 'undefined' || timeout == null) { timeout = 45; }
+                if (typeof success === 'undefined' ) { success = null; }
+                if (typeof error === 'undefined' ) { error = null; }
+                if (this.connection) this.connection.ping.ping( jid, success, error, timeout );
+        };
+
+        this.pong = function (ping){
+		    converse.connection.ping.pong(ping);
+	        return true;
+    	};
+
+	    this.registerPongHandler = function (){
+			converse.connection.ping.addPingHandler( this.pong );
+    	};
+
+	    this.registerPingHandler = function (){
+            if (this.ping_interval>0){
+		        window.setInterval( function() { converse.ping(); }, this.ping_interval*1000);
+                }
+    	};
```

Comment:
`addTimedHandler` doesn't seem to be doing what I have in mind though. I'm thinking of something that does an action only after there has been X amount of time where no stanzas were sent.